stdlib/public/Reflection/Sources/_Runtime/ImageInspection.swift

```diff
@@ -0,0 +1,97 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+@available(SwiftStdlib 5.9, *)
+@frozen
+public enum ImageInspection {}
+
+@_silgen_name("snapshot")
+func snapshot()
+
+//===----------------------------------------------------------------------===//
+// Runtime Attributes
+//===----------------------------------------------------------------------===//
+
+@available(SwiftStdlib 5.9, *)
+var attributeCache: Lock<[ContextDescriptor: [UnsafeRawPointer]]> = .create(
```

Comment:
Is this cache actually needed at this point? The cost of walking all the images can't be that much, can it? Could `snapshot()` just return a copy of the cache (on Darwin) or a copy of the output of `swift_enumerateAllMetadataSections()` (elsewhere)?